src/wp-admin/js/dashboard.js

```diff
@@ -496,4 +496,35 @@ jQuery( function( $ ) {
 			}
 		});
 	}
-});
+	
+	/*
+	* Toggle the CP Petitions Dashboard Widget.
+	*/
+	var tabs = $('ul.petitions-tabs > li');
+	var tabpanes = $('.petitions-pane');
+
+	//add active classes to first tab and petitions-content pane
+	tabs[0].classList.add('active');
+	tabpanes[0].classList.add('active');
+
+	for (var i = 0; i < tabs.length; i++) {
+		tabs[i].addEventListener('click', switchTab);
+	}
+
+	function switchTab(event) {
+			
+		event.preventDefault();
+
+		//Remove active classes to first tab and petitions-content pane
+		tabs.removeClass('active');
+		tabpanes.removeClass('active');
+		
+		var clickedTab = event.currentTarget;
+		var anchor = event.target;
+		var activePaneID = anchor.getAttribute('href');
+		
+		// Get petitions-pane with same ID & add .active
+		clickedTab.classList.add('active');
+		$('#'+ activePaneID).addClass('active');
+	}
+});
```

Comment:
As far as the basic structure of this functionality, there are two ways to go:

1. Load the petitions data from the browser using an AJAX request to the API endpoint.  Make sure an appropriate message is shown with JavaScript disabled.
2. Load the petitions data on the server using `wp_remote_get`.  Make sure the tabs work with JavaScript disabled.

A benefit of using approach (2) is that the functionality *can* work using progressive enhancement with JavaScript enabled.  So, let's take advantage of this benefit.  It can work something like this:

- The active tab is determined by `$_GET['petitions_tab']`.  The CSS classes for active tab and hidden/visible content are assigned appropriately (and `empty( $_GET['petitions_tab'] )` means the first tab is shown).
- The tabs are structured using `<a>` elements which point to URLs like `?petitions_tab=TABNAME` so that they work without JavaScript enabled.
- The JavaScript overrides these default URLs to show a new content pane and mark a different tab as active.